internal/mount/mount.go

```diff
@@ -0,0 +1,141 @@
+package mount
+
+import (
+	"fmt"
+	"os"
+	"strings"
+	"sync"
+
+	"git.sr.ht/~spc/go-log"
+	"github.com/openshift/assisted-installer-agent/src/util"
+	"github.com/project-flotta/flotta-operator/models"
+	"golang.org/x/sys/unix"
+)
+
+const (
+	filesystemsFile = "/etc/filesystems"
+)
+
+type Manager struct {
+	lock sync.Mutex
+	dep  util.IDependencies
+
+	// filesystems holds the content of /etc/filesystems
+	// it used to validate the mount type
+	filesystems string
+}
+
+func New() (*Manager, error) {
+	content, err := os.ReadFile(filesystemsFile)
+	if err != nil {
+		log.Warnf("Cannot list content of '%s': %s", filesystemsFile, err)
+		return nil, fmt.Errorf("cannot list content of '%s': %s", filesystemsFile, err)
+	}
+
+	return &Manager{
+		filesystems: string(content),
+		dep:         util.NewDependencies("/"),
+	}, nil
+}
+
+func (m *Manager) Init(config models.DeviceConfigurationMessage) error {
+	return m.Update(config)
+}
+
+func (m *Manager) Update(config models.DeviceConfigurationMessage) error {
+	m.lock.Lock()
+	defer m.lock.Unlock()
+
+	_, currentMounts, err := GetMounts(m.dep)
```

Comment:
Normally, this should be called only when the configuration changed.
https://github.com/project-flotta/flotta-device-worker/blob/main/internal/configuration/configuration.go#129
Or I'm missing something?

Reply:
you're right...I somehow remembered that only in workload-manager we check for diff.
while we do have that logic even before that in configuration-manager:
```
shouldUpdate := !(configurationEqual && workloadsEqual && secretsEqual)
```
so something feels strange here:
we don't have any sync between mounting to running of the workloads, right?
there is a chance that we'll try to:
- run a workload first (by workload-manager)
- fail for unmounted device (by workload-manager)
- mount the device (by mount-manager)

which lead to another question if to unmount device with force or not.
this is a concern of in which order observers act. the device/mount needs to be ready before the workload act on it.

Reply:
it does. Everything is decoupled. We don't know if the mount is for a running workload, we execute a workload even if the mount failed and best of all, we need to add mount manager before workload in the `Observer` list cause, otherwise we risk starting a workload without mounting the device first.
this logic needs to be refactored

Reply:
I feel that the right logic is:
- check if a running workload needs a new mount
- stop the workload if this is the case
- check if a running workload already binds the mounting point and if, yes, report to user that he has to change the mounting point. Remove that mount from the configuration.
- mount devices
- if a mount fails remove it from the stored configuration and report to the user. Otherwise, there will always be a difference between present and future configurations and the device-worker will keep trying to mount devices
- for every successful mount, start the respective workload

Do you see anything else?

Reply:
Now on workload we check if the mount need to be in there, it mounts, if not it skips the thing.